Nuake/src/Rendering/Camera.cpp

```diff
@@ -70,17 +86,24 @@ namespace Nuake
 		return m_Perspective;
 	}
 
+	void Camera::SetTransform(const Matrix4& transform)
+	{
+		m_View = transform;
+		
+	}
+
 	Matrix4 Camera::GetTransform()
 	{
-		glm::mat4 tr = lookAt(Translation, Translation + Direction, Vector3(0, 1, 0));
+		return m_View;
+		glm::mat4 tr = lookAt(Translation, Translation + glm::normalize(Direction), Vector3(0, 1, 0));
 		return tr;
```

Comment:
is this return statement reachable? there seems to have 2 of them in this method.